apps/crn-server/src/data-providers/contentful/analytics.data-provider.ts

```diff
@@ -0,0 +1,141 @@
+                    .flatMap(flattenWorkingGroupLeaders),
+              ),
+            ),
+            workingGroupMemberCount: getUniqueIdCount(
+              team.linkedFrom?.teamMembershipCollection?.items.flatMap(
+                (item) =>
+                  item?.linkedFrom?.usersCollection?.items
+                    .filter(filterOutAlumni)
+                    .flatMap(flattenWorkingGroupMember),
+              ),
+            ),
+            workingGroupPreviousLeadershipRoleCount: getUniqueIdCount(
+              team.linkedFrom?.teamMembershipCollection?.items.flatMap(
+                (item) =>
+                  item?.linkedFrom?.usersCollection?.items
+                    .filter(filterAlumni)
+                    .flatMap(flattenWorkingGroupLeaders),
+              ),
+            ),
+            workingGroupPreviousMemberCount: getUniqueIdCount(
+              team.linkedFrom?.teamMembershipCollection?.items.flatMap(
+                (item) =>
+                  item?.linkedFrom?.usersCollection?.items
+                    .filter(filterAlumni)
+                    .flatMap(flattenWorkingGroupMember),
+              ),
+            ),
+          })) || [],
+    };
+  }
+}
+
+type Team = NonNullable<
+  NonNullable<
+    FetchAnalyticsTeamLeadershipQuery['teamsCollection']
+  >['items'][number]
+>;
+type User = NonNullable<
+  NonNullable<
+    NonNullable<
+      NonNullable<
+        NonNullable<
+          NonNullable<
+            NonNullable<
+              FetchAnalyticsTeamLeadershipQuery['teamsCollection']
+            >['items'][number]
+          >['linkedFrom']
+        >['teamMembershipCollection']
+      >['items'][number]
+    >['linkedFrom']
+  >['usersCollection']
+>['items'][number];
+
+// removes alumnis, assumes any date in the future also make them an alumni
+const filterOutAlumni = (user: User) => user?.alumniSinceDate === null;
+const filterAlumni = (user: User) => user?.alumniSinceDate !== null;
+
+const flattenWorkingGroupLeaders = (user: User): (string | undefined)[] =>
+  user?.linkedFrom?.workingGroupLeadersCollection?.items.flatMap(
+    (workingGroupLeader) =>
+      workingGroupLeader?.linkedFrom?.workingGroupsCollection?.items.map(
+        (item) => item?.sys.id,
+      ),
+  ) || [];
+const flattenInterestGroupLeaders = (user: User): (string | undefined)[] =>
+  user?.linkedFrom?.interestGroupLeadersCollection?.items.flatMap(
+    (interestGroupLeader) =>
+      interestGroupLeader?.linkedFrom?.interestGroupsCollection?.items.map(
+        (item) => item?.sys.id,
+      ),
+  ) || [];
+const flattenWorkingGroupMember = (user: User): (string | undefined)[] =>
+  user?.linkedFrom?.workingGroupMembersCollection?.items.flatMap(
+    (workingGroupMember) =>
+      workingGroupMember?.linkedFrom?.workingGroupsCollection?.items.map(
+        (item) => item?.sys.id,
+      ),
+  ) || [];
+const getUniqueIdCount = (arr: (string | undefined)[] | undefined): number =>
```

Comment:
I think we need to filter `undefined` first so it doesn't add to the count
![Screenshot 2024-03-05 at 07 41 57](https://github.com/yldio/asap-hub/assets/16595804/c5482fdc-153f-4e8d-b291-b6c1b454dd75)